### PR TITLE
Add and organize docs

### PR DESCRIPTION
**Pull request by samuelallan72**
- move user documentation to snapcraft listing
- add security docs
- expand on user and developer docs

### CODE REVIEW

**Review by chanchiwai-ray — APPROVED** [review]

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -5,12 +5,76 @@ license: BSD-3-Clause
 grade: stable
 confinement: strict
 contact: solutions-engineering@lists.canonical.com
-summary: DERP Relay Server for Tailscale
+issues: https://github.com/canonical/derper-snap/issues
```
Comment (Deezzir) [inline snap/snapcraft.yaml:8]: I believe you need to add the source code label as well; see https://github.com/canonical/charmed-openstack-exporter-snap/blob/6096b2fda9b29ad6285a2f7bc0c387c80f59efb8/snap/snapcraft.yaml#L123

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -5,12 +5,76 @@ license: BSD-3-Clause
...
+  * `stun-port`: The UDP port on which to serve STUN. The listener is bound to the same IP (if any) as specified in the `a` option. (default if unset: "3478")
+
+  The config options correspond to the similarly named command line flags for `derper`.
+
+  Options that don't have support to be configured through the snap will be left on their default.
+  See the `derper` command source code at https://github.com/tailscale/tailscale/tree/main/cmd/derper for more technical information on options.
+
+  Example of setting an option:
+
+  ```bash
```
Comment (Deezzir) [inline snap/snapcraft.yaml:42]: There are no code blocks in their markdown; please refer to one of my docs for either dmg, smart-exporter or charmed-openstack-exporter.  As I said before, the syntax is really really lean. You can test your docs by going to the listing and temporarily inserting it in the listing's description(preview won't work, so you will have to save it first)

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -5,12 +5,76 @@ license: BSD-3-Clause
 grade: stable
 confinement: strict
 contact: solutions-engineering@lists.canonical.com
-summary: DERP Relay Server for Tailscale
+issues: https://github.com/canonical/derper-snap/issues
+summary: DERP Server for Tailscale
 description: |
-  Self hosted DERP Relay servers for Tailscale networks.
-  This snap provides a simple way to run your own DERP servers to route your tailscale traffic through your own relay servers.
+  Derper is a self hosted DERP (Designated Encrypted Relay for Packets) server, designed to be used with Tailscale.
```
Comment (Deezzir) [inline snap/snapcraft.yaml:11]: non-blocking: I would prefer following the Diataxis in some form for cohesion between the projects

**Review by Deezzir — CHANGES_REQUESTED** [review]

File: snap/snapcraft.yaml (hunk 2 above)
Comment (samuelallan72) [reply snap/snapcraft.yaml:42]: Ah of course, indenting rather than backticks :+1:

File: snap/snapcraft.yaml (hunk 3 above)
Comment (jneo8) [reply snap/snapcraft.yaml:12]: In my view, the current description might be too lengthy.

The snapcraft.yaml description appears when users run snap info <snap-name>. After reviewing some popular snaps like MicroK8s, Terraform, and LXD, I suggest condensing the content here and focusing on the essentials.

Additional details could be moved to a separate document for users who need more in-depth information.

**Review by jneo8 — COMMENTED** [review]
Leave comment.

File: snap/snapcraft.yaml (hunk 3 above)
Comment (samuelallan72) [reply snap/snapcraft.yaml:12]: Agreed, it's a lot in one place. Do you have suggestions for where to move the additional details? The snap store doesn't have an equivalent to discourse based docs like charmhub.

Comment (chanchiwai-ray) [reply snap/snapcraft.yaml:12]: Until snap store has some thing equivalent to discourse page like charmhub, I think it's fine to have summary in the description, and add a link to the github readme (or something like) for detail description following Diataxis in some form

Comment (jneo8) [reply snap/snapcraft.yaml:12]: > Agreed, it's a lot in one place. Do you have suggestions for where to move the additional details? The snap store doesn't have an equivalent to discourse based docs like charmhub.

I more prefer separate document on github. Only put a link on description.

**Review by Deezzir — APPROVED** [review]

File: snap/snapcraft.yaml [hunk 4]
```diff
@@ -5,12 +5,69 @@ license: BSD-3-Clause
...
+  **Security**
+
+  Derper provides a way to negotiate direct connections between devices running tailscale,
+  or acting as a full relay server if no direct connection is possible.
+  Derper never sees the content of traffic it relays, as all data is end to end encrypted already by Tailscale using Wireguard on the end devices.
+  Please see the upstream documentation for more security information about derper.
+
+  The snap is strictly confined, with interfaces limited to:
+
+  * `network`: general network access.
```
Comment (Deezzir) [inline snap/snapcraft.yaml:64]: I am not sure if you can use `*` for the list instead of `-`, so I hope you tested it on Snapcraft

Comment (samuelallan72) [reply snap/snapcraft.yaml:64]: Yep I've been following the markdown syntax guide on the snap store listing page:

Comment (samuelallan72) [reply snap/snapcraft.yaml:64]: The **Code** section is not quite accurate though - see https://github.com/canonical/snapcraft.io/issues/4611

**Pull request merged**